pkgs/development/python-modules/outlines-core/default.nix

```diff
@@ -0,0 +1,84 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+  cargo,
+  rustPlatform,
+  rustc,
+  setuptools-rust,
+  setuptools-scm,
+  interegular,
+  jsonschema,
+  datasets,
+  numpy,
+  pytestCheckHook,
+  pydantic,
+  scipy,
+  torch,
+  transformers,
+}:
+
+buildPythonPackage rec {
+  pname = "outlines-core";
+  version = "0.1.17";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "outlines_core";
+    hash = "sha256-CSWhKrHogYxvGJb+79tNTYySLmRhUVBi0b0+YEWbAoE=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
```

Comment:
We should use the new vendoring to prevent commiting the lock file

Reply:
Upstream doesn't provide one. Could try making an issue, but this might not be avoidable

Reply:
I discussed with upstream, they will make one in future versions.